test/lit/merge/type-imports.wat

```diff
@@ -0,0 +1,133 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+
+;; RUN: wasm-merge %s first %s.second second -all -S -o - | filecheck %s
+
+(module
+  ;; Bound to an abstract heap type
+  (import "second" "t1" (type $t1))
+
+  ;; Bound to a concrete heap type
+  (import "second" "t2" (type $t2))
+
+  ;; Bound to an imported type
+  ;; CHECK:      (import "third" "t3" (type $t3 (sub struct)))
+  (import "second" "t3" (type $t3))
+
+  ;; Left unbound
+  ;; CHECK:      (import "third" "t4" (type $t4 (sub any)))
+  (import "third" "t4" (type $t4))
+
+  ;; Check the import of a function using imported types
+  (import "second" "g" (func $g (param (ref $t2))))
+
+  ;; Check that the function parameters are updated
+  (func (export "f") (param (ref $t1) (ref $t2) (ref $t3) (ref $t4))
```

Comment:
If you give these functions `$names`, then the test output update script should be able to match them up with functions in the output and put them next to each other.